### PR TITLE
SONAR-9092 let api/projects/provisioned use protobuf

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java [hunk 1]
```diff
@@ -23,31 +23,59 @@
...
+import static org.sonar.server.es.SearchOptions.MAX_LIMIT;
 import static org.sonar.server.project.ws.ProjectsWsSupport.PARAM_ORGANIZATION;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
 
 public class ProvisionedAction implements ProjectsWsAction {
 
   private static final Set<String> QUALIFIERS_FILTER = newHashSet(Qualifiers.PROJECT);
-  private static final Set<String> POSSIBLE_FIELDS = newHashSet("uuid", "key", "name", "creationDate");
+
+  private static enum PossibleField {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: MINOR Remove this redundant "static" qualifier. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2786)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: Why do you need an enum when a list of string is enough ?

File: sonar-ws/src/main/protobuf/ws-components.proto [hunk 2]
```diff
@@ -87,6 +87,21 @@ message SearchProjectsWsResponse {
   optional sonarqube.ws.commons.Facets facets = 3;
 }
 
+// WS api/components/provisioned
+message ProvisionedWsResponse {
+  repeated Component projects = 1;
+  optional int64 total = 2;
```
Comment (julienlancelot) [inline sonar-ws/src/main/protobuf/ws-components.proto]: This is the old paging way, the new way is to use : 
`optional sonarqube.ws.commons.Paging paging = 1;`
(you need to add another entry in the changelog)

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: The enum has been removed.

Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: Done.

File: sonar-ws/src/main/protobuf/ws-components.proto (hunk 2 above)
Comment (bartfastiel) [inline sonar-ws/src/main/protobuf/ws-components.proto]: Cool. I added another commit to use the new pagination.

File: server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java [hunk 3]
```diff
@@ -82,7 +83,10 @@ public void define(WebService.NewController controller) {
       .addSearchQuery("sonar", "names", "keys")
       .addFieldsParam(POSSIBLE_FIELDS);
 
-    action.setChangelog(new Change("6.4", "The 'uuid' field is deprecated in the response"));
+    action.setChangelog(
+      new Change("6.4", "The 'uuid' field is deprecated in the response"),
+      new Change("6.4", "The fields 'total', 'p' and 'ps' are now named 'total', 'pageIndex' and 'pageSize' in the response " +
+        "and are nested inside an attribute 'paging'"));
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: You can use a simpler sentence like "Paging response fields is now in a Paging object"

Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/project/ws/ProvisionedAction.java]: Done.

**Pull request merged**